src/Collection.php

```diff
@@ -89,17 +88,73 @@ protected function loop() {
 	/**
 	 * This function allow to format every item that is returned to the endpoint
 	 * the filter sends 3 params to the user so can be more easy to manipulate the
-	 * data based on certaim params.
+	 * data based on certain params.
 	 *
-	 * @param \WP_Post $the_post Current post object.
 	 * @return array The formated data from every item.
 	 */
-	protected function format_item( \WP_Post $the_post ) {
-		$item = [ 'id' => $the_post->ID ];
-		if ( $this->format_item ) {
-			$item = apply_filters( Filter::ITEM_FORMAT, $item, $the_post, $this->args );
+	protected function format_item() {
+		$the_post = get_post();
+
+		// Author.
+		$the_author = get_userdata( $the_post->post_author );
+
+		// Thumbnail.
+		if ( has_post_thumbnail( $the_post->ID ) ) {
+			$size = apply_filters( Filter::THUMBNAIL_SIZE, 'thumbnail', $the_post );
```

Comment:
I think this filter can use the ID in order to be specific of the post like:

```
ln_endpoints_collection_thumbnail_size_2
```

Or we  should pass the `$this->args` instead since the queries are around the args and not the post, so is easily to apply the same filter in a collection of posts once.